src/printer.rs

```diff
@@ -86,11 +86,20 @@ impl Printer {
         }
     }
 
-    fn print_json_text(&mut self, text: &str) -> io::Result<()> {
+    fn print_json_text(&mut self, text: &str, check_valid: bool) -> io::Result<()> {
         if !self.indent_json {
             // We don't have to do anything specialized, so fall back to the generic version
-            self.print_syntax_text(text, "json")
-        } else if self.color {
+            return self.print_syntax_text(text, "json");
+        }
+
+        if check_valid && !valid_json(text) {
+            // JSONXF may mess up the text, e.g. by removing whitespace
+            // This is somewhat common as application/json is the default
+            // content type for requests
+            return self.print_syntax_text(text, "json");
+        }
```

Comment:
Can we move this check to when pattern-matching the content-type?

Reply:
I thought about that, but this way we can skip the JSON check if formatting isn't necessary.